lib/core/bucketEntries/usecase.ts

```diff
@@ -0,0 +1,51 @@
+import { BucketsRepository } from '../buckets/Repository';
+import { BucketEntriesRepository } from './Repository';
+import { BucketNotFoundError, BucketForbiddenError, BucketEntryNotFoundError } from '../buckets/usecase';
+import { FramesRepository } from '../frames/Repository';
+import { PointersRepository } from '../pointers/Repository';
+
+export class BucketEntriesUsecase {
```

Comment:
> As stated in previous PRs, maybe we can get this `removeFile` and pass it to TypeScript on a bucket entries use case?

I've added removeFile in BucketsEntiesUsecase wonder if perhaps we should add it on BucketsUsecase? 

I still have some questions. I see we have in many repositories `deleteByIds` maybe we should also create `deletebyId` in singular or shall we just use an array with one id as in this PR?

Shall we force to enqueue a message for every file deletion? or shall we keep the structure in services/files.js and decouple them?

Still haven't deleted services/files.js for reference. Will delete it as soon as this questions are solved.

Reply:
converted to draft until @sg-gs reviews these questions

Reply:
- When a bulk delete with a known size is available, is preferred to delete things in batches of 10 than performing 10 queries of one delete at the same time. 

- The case of passing just one id inside an array is not a bad idea, I have not seen any performance or detrimental issues of doing that which the cleaning scripts, which have to go as fast as possible without affecting the database, so no problem with that, could be done

- As we are using a queue and tomorrow could be a table, or who knows, coupling the file deletion to the enqueuing could be a bad deal. Decoupling if possible seems better, allowing a callback on each file deletion completed, for instance.

Reply:
Sounds good, I agree with the explanation. We will have a callback `beforePointerIsDeleted` then

Should we create another usecase for this logic? or where do you think it should live?:

```
  const { Mirror } = storage.models;

  if (!Mirror) {
    throw new Error('Missing Mirror model');
  }
  const { hash } = pointer;

  const mirrors = await Mirror.find({ shardHash: hash }).populate('contact').exec();
  const stillExistentMirrors = mirrors.filter((mirror) => {
    return mirror.contact && mirror.contact.address && mirror.contact.port;
  });

  for (const mirror of stillExistentMirrors) {
    const { address, port } = mirror.contact;

    let url = `http://${address}:${port}/shards/${hash}`;

    if (bucketEntry.version === 2) {
      url = `http://${address}:${port}/v2/shards/${hash}`;
    }

    await new Promise((resolve, reject) => {
      networkQueue.enqueueMessage({
        type: DELETING_FILE_MESSAGE,
        payload: { hash, url }
      }, (err) => {
        if (err) {
          log.error(
            'deletePointer: Error enqueueing pointer %s shard %s deletion task: %s',
            pointer._id,
            pointer.shard,
            err.message
          );
          // We do not reject the error since if we fail to enqueue the message, we don't want requests to fail.
          // reject(err);
        } else {
          resolve();
        }
      });
    });
  }
```

Reply:
Is a bit troublesome in this case because a lot of models are involved, as we are moving from v1 to v2, take into count that Pointer and Frame disappear. 

That changes mean that at the end of the migration, all the v1 logic will disappear and the efforts to keep that logic located could be less or none, giving priority to v2. In short, from my point of view, I will locate it in Shards' use case due to the fact that mainly the Shard and Mirror entities are involved here. 

Maybe, providing an array of Shards to delete as a parameter and giving the function the responsibility of looking for their mirrors and deleting everything, could be a decent, scalable solution

Reply:
sounds reasonable! will do 👌